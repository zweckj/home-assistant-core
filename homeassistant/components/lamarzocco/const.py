"""Constants for the La Marzocco integration."""

from typing import Final

DOMAIN: Final = "lamarzocco"

POLLING_INTERVAL: Final = 30
UPDATE_DELAY: Final = 3

DEFAULT_CLIENT_ID: Final = "7_1xwei9rtkuckso44ks4o8s0c0oc4swowo00wgw0ogsok84kosg"
DEFAULT_CLIENT_SECRET: Final = "2mgjqpikbfuok8g4s44oo4gsw0ks44okk4kc4kkkko0c8soc8s"

DEFAULT_PORT_LOCAL: Final = 8081

MODEL_GS3_AV: Final = "GS3 AV"
MODEL_GS3_MP: Final = "GS3 MP"
MODEL_LM: Final = "Linea Mini"
MODEL_LMU: Final = "Micra"

BREW_ACTIVE: Final = "brew_active"

GLOBAL: Final = "global"
MON: Final = "mon"
TUE: Final = "tue"
WED: Final = "wed"
THU: Final = "thu"
FRI: Final = "fri"
SAT: Final = "sat"
SUN: Final = "sun"

DAYS: Final = [MON, TUE, WED, THU, FRI, SAT, SUN]

MACHINE_NAME: Final = "machine_name"
SERIAL_NUMBER: Final = "serial_number"
