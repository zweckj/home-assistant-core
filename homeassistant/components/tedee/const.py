"""Constants for the Tedee integration."""
from datetime import timedelta

DOMAIN = "tedee"
NAME = "Tedee"

SCAN_INTERVAL = timedelta(seconds=10)

CONF_LOCAL_ACCESS_TOKEN = "local_access_token"
CONF_USE_LOCAL_API = "use_local_api"
CONF_USE_CLOUD_API = "use_cloud_api"
CONF_CLOUD_BRIDGE_ID = "cloud_bridge_id"
