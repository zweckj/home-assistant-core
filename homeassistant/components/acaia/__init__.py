"""Initialize the Acaia component."""
from homeassistant.config_entries import ConfigEntry
from homeassistant.const import CONF_MAC, CONF_NAME, Platform
from homeassistant.core import HomeAssistant
from homeassistant.helpers import config_validation as cv
from homeassistant.helpers.typing import ConfigType

from .acaiaclient import AcaiaClient
from .const import CONF_IS_NEW_STYLE_SCALE, DOMAIN
from .coordinator import AcaiaApiCoordinator

CONFIG_SCHEMA = cv.config_entry_only_config_schema(DOMAIN)

PLATFORMS = [Platform.BUTTON, Platform.SENSOR]


async def async_setup(hass: HomeAssistant, config: ConfigType) -> bool:
    """Set up the Acaia component."""
    hass.data.setdefault(DOMAIN, {})
    return True


async def async_setup_entry(hass: HomeAssistant, config_entry: ConfigEntry) -> bool:
    """Set up Acaia as config entry."""

    name = config_entry.data[CONF_NAME]
    mac = config_entry.data[CONF_MAC]
    is_new_style_scale = config_entry.data.get(CONF_IS_NEW_STYLE_SCALE, True)

    acaia_client = AcaiaClient(
        hass, mac=mac, name=name, is_new_style_scale=is_new_style_scale
    )

    hass.data[DOMAIN][config_entry.entry_id] = coordinator = AcaiaApiCoordinator(
        hass, acaia_client
    )

    await coordinator.async_config_entry_first_refresh()

    await hass.config_entries.async_forward_entry_setups(config_entry, PLATFORMS)

    return True


async def async_unload_entry(hass: HomeAssistant, config_entry: ConfigEntry) -> bool:
    """Unload a config entry."""

    unload_ok = await hass.config_entries.async_unload_platforms(
        config_entry, PLATFORMS
    )

    if unload_ok:
        hass.data[DOMAIN].pop(config_entry.entry_id)

    return unload_ok
