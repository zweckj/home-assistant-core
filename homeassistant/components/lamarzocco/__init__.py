"""The La Marzocco integration."""

import logging

from lmcloud.client_cloud import LaMarzoccoCloudClient
from lmcloud.const import MachineModel
from lmcloud.exceptions import AuthFail, RequestNotSuccessful

from homeassistant.config_entries import ConfigEntry
from homeassistant.const import (
    CONF_HOST,
    CONF_MODEL,
    CONF_NAME,
    CONF_PASSWORD,
    CONF_TOKEN,
    CONF_USERNAME,
    Platform,
)
from homeassistant.core import HomeAssistant
from homeassistant.exceptions import ConfigEntryError

from .const import DOMAIN
from .coordinator import LaMarzoccoMachineUpdateCoordinator

PLATFORMS = [
    Platform.BINARY_SENSOR,
    Platform.BUTTON,
    Platform.NUMBER,
    Platform.SELECT,
    Platform.SENSOR,
    Platform.SWITCH,
    Platform.UPDATE,
]

_LOGGER = logging.getLogger(__name__)


async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
    """Set up La Marzocco as config entry."""

    if (model := entry.data[CONF_MODEL]) in MachineModel:
        coordinator = LaMarzoccoMachineUpdateCoordinator(hass)
    else:
        raise ConfigEntryError(f"Invalid model {model}")

    await coordinator.async_config_entry_first_refresh()
    hass.data.setdefault(DOMAIN, {})[entry.entry_id] = coordinator

    await hass.config_entries.async_forward_entry_setups(entry, PLATFORMS)

    return True


async def async_unload_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
    """Unload a config entry."""

    unload_ok = await hass.config_entries.async_unload_platforms(entry, PLATFORMS)

    if unload_ok:
        hass.data[DOMAIN].pop(entry.entry_id)

    return unload_ok


async def async_migrate_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
    """Migrate config entry."""
    if entry.version == 1:
        cloud_client = LaMarzoccoCloudClient(
            username=entry.data[CONF_USERNAME],
            password=entry.data[CONF_PASSWORD],
        )
        try:
            fleet = await cloud_client.get_customer_fleet()
        except (AuthFail, RequestNotSuccessful) as exc:
            raise ConfigEntryError("Migration failed") from exc

        assert entry.unique_id
        device = fleet[entry.unique_id]
        v2_data = {
            CONF_USERNAME: entry.data[CONF_USERNAME],
            CONF_PASSWORD: entry.data[CONF_PASSWORD],
            CONF_MODEL: device.model,
            CONF_NAME: device.name,
            CONF_TOKEN: device.communication_key,
        }

        if CONF_HOST in entry.data:
            v2_data[CONF_HOST] = entry.data[CONF_HOST]

        entry.version = 2
        hass.config_entries.async_update_entry(
            entry,
            data=v2_data,
        )
        _LOGGER.debug("Migrated La Marzocco config entry to version 2")
    return True
