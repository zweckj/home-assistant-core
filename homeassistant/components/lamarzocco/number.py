"""Water heater platform for La Marzocco espresso machines."""

from collections.abc import Callable, Coroutine
from dataclasses import dataclass
from typing import Any

from lmcloud.const import LaMarzoccoModel

from homeassistant.components.number import (
    NumberDeviceClass,
    NumberEntity,
    NumberEntityDescription,
)
from homeassistant.config_entries import ConfigEntry
from homeassistant.const import PRECISION_TENTHS, UnitOfTemperature
from homeassistant.core import HomeAssistant
from homeassistant.helpers.entity_platform import AddEntitiesCallback

from .const import DOMAIN
from .coordinator import LaMarzoccoUpdateCoordinator
from .entity import LaMarzoccoEntity, LaMarzoccoEntityDescription


@dataclass(frozen=True, kw_only=True)
class LaMarzoccoNumberEntityDescription(
    LaMarzoccoEntityDescription,
    NumberEntityDescription,
):
    """Description of an La Marzocco number entity."""

    native_value_fn: Callable[[LaMarzoccoUpdateCoordinator], float | int]
    set_value_fn: Callable[
        [LaMarzoccoUpdateCoordinator, float | int], Coroutine[Any, Any, bool]
    ]


ENTITIES: tuple[LaMarzoccoNumberEntityDescription, ...] = (
    LaMarzoccoNumberEntityDescription(
        key="coffee_temp",
        translation_key="coffee_temp",
        icon="mdi:coffee-maker",
        device_class=NumberDeviceClass.TEMPERATURE,
        native_unit_of_measurement=UnitOfTemperature.CELSIUS,
        native_step=PRECISION_TENTHS,
        native_min_value=85,
        native_max_value=104,
        set_value_fn=lambda coordinator, temp: coordinator.lm.set_coffee_temp(temp),
        native_value_fn=lambda coordinator: coordinator.lm.current_status.get(
            "coffee_set_temp", 0
        ),
    ),
    LaMarzoccoNumberEntityDescription(
        key="steam_temp",
        translation_key="steam_temp",
        icon="mdi:kettle-steam",
        device_class=NumberDeviceClass.TEMPERATURE,
        native_unit_of_measurement=UnitOfTemperature.CELSIUS,
        native_step=PRECISION_TENTHS,
        native_min_value=126,
        native_max_value=131,
        set_value_fn=lambda coordinator, temp: coordinator.lm.set_steam_temp(
            round(temp)
        ),
        native_value_fn=lambda coordinator: coordinator.lm.current_status.get(
            "steam_set_temp", 0
        ),
        supported_models=(
            LaMarzoccoModel.GS3_AV,
            LaMarzoccoModel.GS3_MP,
        ),
    ),
)


async def async_setup_entry(
    hass: HomeAssistant,
    config_entry: ConfigEntry,
    async_add_entities: AddEntitiesCallback,
) -> None:
    """Set up water heater type entities."""
    coordinator = hass.data[DOMAIN][config_entry.entry_id]

    async_add_entities(
        LaMarzoccoNumberEntity(coordinator, description)
        for description in ENTITIES
        if coordinator.lm.model_name in description.supported_models
    )


class LaMarzoccoNumberEntity(LaMarzoccoEntity, NumberEntity):
    """Water heater representing espresso machine temperature data."""

    entity_description: LaMarzoccoNumberEntityDescription

    @property
    def native_value(self) -> float:
        """Return the current value."""
        return self.entity_description.native_value_fn(self.coordinator)

    async def async_set_native_value(self, value: float) -> None:
        """Set the value."""
        await self.entity_description.set_value_fn(self.coordinator, value)
        self.async_write_ha_state()
