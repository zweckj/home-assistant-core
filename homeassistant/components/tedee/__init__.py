"""Init the tedee component."""

from collections.abc import Awaitable, Callable
from http import HTTPStatus
import logging
import secrets
import string
from typing import Any

from aiohttp.hdrs import METH_POST
from aiohttp.web import Request, Response
from pytedee_async.exception import TedeeWebhookException

from homeassistant.components.http import HomeAssistantView
from homeassistant.components.webhook import (
    async_generate_path as webhook_generate_path,
    async_register as webhook_register,
    async_unregister as webhook_unregister,
)
from homeassistant.config_entries import ConfigEntry
from homeassistant.const import CONF_WEBHOOK_ID, EVENT_HOMEASSISTANT_STOP, Platform
from homeassistant.core import HomeAssistant
from homeassistant.helpers import device_registry as dr
from homeassistant.helpers.network import get_url

from .const import DOMAIN, NAME
from .coordinator import TedeeApiCoordinator

PLATFORMS = [
    Platform.BINARY_SENSOR,
    Platform.LOCK,
    Platform.SENSOR,
]

_LOGGER = logging.getLogger(__name__)


async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
    """Integration setup."""

    coordinator = TedeeApiCoordinator(hass)

    await coordinator.async_config_entry_first_refresh()

    device_registry = dr.async_get(hass)
    device_registry.async_get_or_create(
        config_entry_id=entry.entry_id,
        identifiers={(DOMAIN, coordinator.bridge.serial)},
        manufacturer="Tedee",
        name=coordinator.bridge.name,
        model="Bridge",
        serial_number=coordinator.bridge.serial,
    )

    hass.data.setdefault(DOMAIN, {})[entry.entry_id] = coordinator

    def generate_api_key(length: int = 16) -> str:
        characters = string.ascii_letters + string.digits
        password = "".join(secrets.choice(characters) for _ in range(length))
        return password

    async def unregister_webhook(_: Any) -> None:
        await coordinator.async_unregister_webhook()
        webhook_unregister(hass, entry.data[CONF_WEBHOOK_ID])

    async def register_webhook() -> None:
        # webhook_url = webhook_generate_url(hass, entry.data[CONF_WEBHOOK_ID])
        webhook_url = (
            f"http://192.168.1.114:8123/api/webhook/{entry.data[CONF_WEBHOOK_ID]}"
        )
        webhook_url = f"{get_url(hass, allow_external=False)}{webhook_generate_path(entry.data[CONF_WEBHOOK_ID])}"
        webhook_name = "Tedee"
        if entry.title != NAME:
            webhook_name = f"{NAME} {entry.title}"

        api_key = generate_api_key()

        webhook_register(
            hass,
            DOMAIN,
            webhook_name,
            entry.data[CONF_WEBHOOK_ID],
            get_webhook_handler(coordinator, api_key),
            allowed_methods=[METH_POST],
        )
        _LOGGER.warning("Registered Tedee webhook at hass: %s", webhook_url)

        try:
            await coordinator.async_register_webhook(
                webhook_url, headers=[{"Authorization": api_key}]
            )
        except TedeeWebhookException as ex:
            _LOGGER.warning("Failed to register Tedee webhook from bridge: %s", ex)
        else:
            entry.async_on_unload(
                hass.bus.async_listen_once(EVENT_HOMEASSISTANT_STOP, unregister_webhook)
            )

    entry.async_create_background_task(
        hass, register_webhook(), "tedee_register_webhook"
    )
    await hass.config_entries.async_forward_entry_setups(entry, PLATFORMS)

    return True


async def async_unload_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
    """Unload a config entry."""

    if unload_ok := await hass.config_entries.async_unload_platforms(entry, PLATFORMS):
        hass.data[DOMAIN].pop(entry.entry_id)

    return unload_ok


def get_webhook_handler(
    coordinator: TedeeApiCoordinator,
    api_key: str,
) -> Callable[[HomeAssistant, str, Request], Awaitable[Response | None]]:
    """Return webhook handler."""

    async def async_webhook_handler(
        hass: HomeAssistant, webhook_id: str, request: Request
    ) -> Response | None:
        # Handle http post calls to the path.
        if not request.body_exists:
            return HomeAssistantView.json(
                result="No Body", status_code=HTTPStatus.BAD_REQUEST
            )

        if request.headers.get("Authorization") != api_key:
            return HomeAssistantView.json(
                result="Unauthorized", status_code=HTTPStatus.UNAUTHORIZED
            )

        body = await request.json()
        try:
            coordinator.webhook_received(body)
        except TedeeWebhookException as ex:
            return HomeAssistantView.json(
                result=str(ex), status_code=HTTPStatus.BAD_REQUEST
            )

        return HomeAssistantView.json(result="OK", status_code=HTTPStatus.OK)

    return async_webhook_handler
