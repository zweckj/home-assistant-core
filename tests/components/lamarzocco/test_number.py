"""Tests for the La Marzocco Water Heaters."""


from unittest.mock import MagicMock

from lmcloud.const import LaMarzoccoModel
import pytest
from syrupy import SnapshotAssertion

from homeassistant.components.number import (
    ATTR_VALUE,
    DOMAIN as NUMBER_DOMAIN,
    SERVICE_SET_VALUE,
)
from homeassistant.const import ATTR_ENTITY_ID
from homeassistant.core import HomeAssistant
from homeassistant.helpers import entity_registry as er

pytestmark = pytest.mark.usefixtures("init_integration")


async def test_coffee_boiler(
    hass: HomeAssistant,
    mock_lamarzocco: MagicMock,
    entity_registry: er.EntityRegistry,
    snapshot: SnapshotAssertion,
) -> None:
    """Test the La Marzocco coffee temperature Number."""
    serial_number = mock_lamarzocco.serial_number

    mock_lamarzocco.set_power.return_value = None
    mock_lamarzocco.set_coffee_temp.return_value = None

    state = hass.states.get(f"number.{serial_number}_coffee_temperature")

    assert state
    assert state == snapshot

    entry = entity_registry.async_get(state.entity_id)
    assert entry
    assert entry.device_id
    assert entry == snapshot

    # on/off service calls
    await hass.services.async_call(
        NUMBER_DOMAIN,
        SERVICE_SET_VALUE,
        {
            ATTR_ENTITY_ID: f"number.{serial_number}_coffee_temperature",
            ATTR_VALUE: 95,
        },
        blocking=True,
    )

    assert len(mock_lamarzocco.set_coffee_temp.mock_calls) == 1
    mock_lamarzocco.set_coffee_temp.assert_called_once_with(temperature=95)


@pytest.mark.parametrize(
    "device_fixture", [LaMarzoccoModel.GS3_AV, LaMarzoccoModel.GS3_MP]
)
async def test_steam_boiler(
    hass: HomeAssistant,
    mock_lamarzocco: MagicMock,
    entity_registry: er.EntityRegistry,
    snapshot: SnapshotAssertion,
) -> None:
    """Test the La Marzocco steam temperature number."""

    serial_number = mock_lamarzocco.serial_number

    state = hass.states.get(f"number.{serial_number}_steam_temperature")
    assert state
    assert state == snapshot

    entry = entity_registry.async_get(state.entity_id)
    assert entry
    assert entry.device_id
    assert entry == snapshot

    # on/off service calls
    await hass.services.async_call(
        NUMBER_DOMAIN,
        SERVICE_SET_VALUE,
        {
            ATTR_ENTITY_ID: f"number.{serial_number}_steam_temperature",
            ATTR_VALUE: 131,
        },
        blocking=True,
    )

    assert len(mock_lamarzocco.set_steam_temp.mock_calls) == 1
    mock_lamarzocco.set_steam_temp.assert_called_once_with(temperature=131)


@pytest.mark.parametrize(
    "device_fixture", [LaMarzoccoModel.LINEA_MICRA, LaMarzoccoModel.LINEA_MINI]
)
async def test_steam_boiler_none(
    hass: HomeAssistant,
    mock_lamarzocco: MagicMock,
    entity_registry: er.EntityRegistry,
    snapshot: SnapshotAssertion,
) -> None:
    """Ensure steam boiler number is None for unsupported models."""

    serial_number = mock_lamarzocco.serial_number

    state = hass.states.get(f"number.{serial_number}_steam_temperature")
    assert state is None
