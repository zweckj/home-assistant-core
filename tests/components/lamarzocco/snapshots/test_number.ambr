# serializer version: 1
# name: test_coffee_boiler[GS3 AV]
  StateSnapshot({
    'attributes': ReadOnlyDict({
      'device_class': 'temperature',
      'friendly_name': 'GS01234 Coffee temperature',
      'icon': 'mdi:coffee-maker',
      'max': 104,
      'min': 85,
      'mode': <NumberMode.AUTO: 'auto'>,
      'step': 0.1,
      'unit_of_measurement': <UnitOfTemperature.CELSIUS: '°C'>,
    }),
    'context': <ANY>,
    'entity_id': 'number.gs01234_coffee_temperature',
    'last_changed': <ANY>,
    'last_updated': <ANY>,
    'state': '95',
  })
# ---
# name: test_coffee_boiler[GS3 AV].1
  EntityRegistryEntrySnapshot({
    'aliases': set({
    }),
    'area_id': None,
    'capabilities': dict({
      'max': 104,
      'min': 85,
      'mode': <NumberMode.AUTO: 'auto'>,
      'step': 0.1,
    }),
    'config_entry_id': <ANY>,
    'device_class': None,
    'device_id': <ANY>,
    'disabled_by': None,
    'domain': 'number',
    'entity_category': None,
    'entity_id': 'number.gs01234_coffee_temperature',
    'has_entity_name': True,
    'hidden_by': None,
    'icon': None,
    'id': <ANY>,
    'name': None,
    'options': dict({
    }),
    'original_device_class': <NumberDeviceClass.TEMPERATURE: 'temperature'>,
    'original_icon': 'mdi:coffee-maker',
    'original_name': 'Coffee temperature',
    'platform': 'lamarzocco',
    'previous_unique_id': None,
    'supported_features': 0,
    'translation_key': 'coffee_temp',
    'unique_id': 'GS01234_coffee_temp',
    'unit_of_measurement': <UnitOfTemperature.CELSIUS: '°C'>,
  })
# ---
# name: test_coffee_boiler[GS3 MP]
  StateSnapshot({
    'attributes': ReadOnlyDict({
      'device_class': 'temperature',
      'friendly_name': 'GS01234 Coffee temperature',
      'icon': 'mdi:coffee-maker',
      'max': 104,
      'min': 85,
      'mode': <NumberMode.AUTO: 'auto'>,
      'step': 0.1,
      'unit_of_measurement': <UnitOfTemperature.CELSIUS: '°C'>,
    }),
    'context': <ANY>,
    'entity_id': 'number.gs01234_coffee_temperature',
    'last_changed': <ANY>,
    'last_updated': <ANY>,
    'state': '95',
  })
# ---
# name: test_coffee_boiler[GS3 MP].1
  EntityRegistryEntrySnapshot({
    'aliases': set({
    }),
    'area_id': None,
    'capabilities': dict({
      'max': 104,
      'min': 85,
      'mode': <NumberMode.AUTO: 'auto'>,
      'step': 0.1,
    }),
    'config_entry_id': <ANY>,
    'device_class': None,
    'device_id': <ANY>,
    'disabled_by': None,
    'domain': 'number',
    'entity_category': None,
    'entity_id': 'number.gs01234_coffee_temperature',
    'has_entity_name': True,
    'hidden_by': None,
    'icon': None,
    'id': <ANY>,
    'name': None,
    'options': dict({
    }),
    'original_device_class': <NumberDeviceClass.TEMPERATURE: 'temperature'>,
    'original_icon': 'mdi:coffee-maker',
    'original_name': 'Coffee temperature',
    'platform': 'lamarzocco',
    'previous_unique_id': None,
    'supported_features': 0,
    'translation_key': 'coffee_temp',
    'unique_id': 'GS01234_coffee_temp',
    'unit_of_measurement': <UnitOfTemperature.CELSIUS: '°C'>,
  })
# ---
# name: test_coffee_boiler[Linea Mini]
  StateSnapshot({
    'attributes': ReadOnlyDict({
      'device_class': 'temperature',
      'friendly_name': 'LM01234 Coffee temperature',
      'icon': 'mdi:coffee-maker',
      'max': 104,
      'min': 85,
      'mode': <NumberMode.AUTO: 'auto'>,
      'step': 0.1,
      'unit_of_measurement': <UnitOfTemperature.CELSIUS: '°C'>,
    }),
    'context': <ANY>,
    'entity_id': 'number.lm01234_coffee_temperature',
    'last_changed': <ANY>,
    'last_updated': <ANY>,
    'state': '95',
  })
# ---
# name: test_coffee_boiler[Linea Mini].1
  EntityRegistryEntrySnapshot({
    'aliases': set({
    }),
    'area_id': None,
    'capabilities': dict({
      'max': 104,
      'min': 85,
      'mode': <NumberMode.AUTO: 'auto'>,
      'step': 0.1,
    }),
    'config_entry_id': <ANY>,
    'device_class': None,
    'device_id': <ANY>,
    'disabled_by': None,
    'domain': 'number',
    'entity_category': None,
    'entity_id': 'number.lm01234_coffee_temperature',
    'has_entity_name': True,
    'hidden_by': None,
    'icon': None,
    'id': <ANY>,
    'name': None,
    'options': dict({
    }),
    'original_device_class': <NumberDeviceClass.TEMPERATURE: 'temperature'>,
    'original_icon': 'mdi:coffee-maker',
    'original_name': 'Coffee temperature',
    'platform': 'lamarzocco',
    'previous_unique_id': None,
    'supported_features': 0,
    'translation_key': 'coffee_temp',
    'unique_id': 'LM01234_coffee_temp',
    'unit_of_measurement': <UnitOfTemperature.CELSIUS: '°C'>,
  })
# ---
# name: test_coffee_boiler[Micra]
  StateSnapshot({
    'attributes': ReadOnlyDict({
      'device_class': 'temperature',
      'friendly_name': 'MR01234 Coffee temperature',
      'icon': 'mdi:coffee-maker',
      'max': 104,
      'min': 85,
      'mode': <NumberMode.AUTO: 'auto'>,
      'step': 0.1,
      'unit_of_measurement': <UnitOfTemperature.CELSIUS: '°C'>,
    }),
    'context': <ANY>,
    'entity_id': 'number.mr01234_coffee_temperature',
    'last_changed': <ANY>,
    'last_updated': <ANY>,
    'state': '95',
  })
# ---
# name: test_coffee_boiler[Micra].1
  EntityRegistryEntrySnapshot({
    'aliases': set({
    }),
    'area_id': None,
    'capabilities': dict({
      'max': 104,
      'min': 85,
      'mode': <NumberMode.AUTO: 'auto'>,
      'step': 0.1,
    }),
    'config_entry_id': <ANY>,
    'device_class': None,
    'device_id': <ANY>,
    'disabled_by': None,
    'domain': 'number',
    'entity_category': None,
    'entity_id': 'number.mr01234_coffee_temperature',
    'has_entity_name': True,
    'hidden_by': None,
    'icon': None,
    'id': <ANY>,
    'name': None,
    'options': dict({
    }),
    'original_device_class': <NumberDeviceClass.TEMPERATURE: 'temperature'>,
    'original_icon': 'mdi:coffee-maker',
    'original_name': 'Coffee temperature',
    'platform': 'lamarzocco',
    'previous_unique_id': None,
    'supported_features': 0,
    'translation_key': 'coffee_temp',
    'unique_id': 'MR01234_coffee_temp',
    'unit_of_measurement': <UnitOfTemperature.CELSIUS: '°C'>,
  })
# ---
# name: test_steam_boiler[GS3 AV]
  StateSnapshot({
    'attributes': ReadOnlyDict({
      'device_class': 'temperature',
      'friendly_name': 'GS01234 Steam temperature',
      'icon': 'mdi:kettle-steam',
      'max': 131,
      'min': 126,
      'mode': <NumberMode.AUTO: 'auto'>,
      'step': 0.1,
      'unit_of_measurement': <UnitOfTemperature.CELSIUS: '°C'>,
    }),
    'context': <ANY>,
    'entity_id': 'number.gs01234_steam_temperature',
    'last_changed': <ANY>,
    'last_updated': <ANY>,
    'state': '128',
  })
# ---
# name: test_steam_boiler[GS3 AV].1
  EntityRegistryEntrySnapshot({
    'aliases': set({
    }),
    'area_id': None,
    'capabilities': dict({
      'max': 131,
      'min': 126,
      'mode': <NumberMode.AUTO: 'auto'>,
      'step': 0.1,
    }),
    'config_entry_id': <ANY>,
    'device_class': None,
    'device_id': <ANY>,
    'disabled_by': None,
    'domain': 'number',
    'entity_category': None,
    'entity_id': 'number.gs01234_steam_temperature',
    'has_entity_name': True,
    'hidden_by': None,
    'icon': None,
    'id': <ANY>,
    'name': None,
    'options': dict({
    }),
    'original_device_class': <NumberDeviceClass.TEMPERATURE: 'temperature'>,
    'original_icon': 'mdi:kettle-steam',
    'original_name': 'Steam temperature',
    'platform': 'lamarzocco',
    'previous_unique_id': None,
    'supported_features': 0,
    'translation_key': 'steam_temp',
    'unique_id': 'GS01234_steam_temp',
    'unit_of_measurement': <UnitOfTemperature.CELSIUS: '°C'>,
  })
# ---
# name: test_steam_boiler[GS3 MP]
  StateSnapshot({
    'attributes': ReadOnlyDict({
      'device_class': 'temperature',
      'friendly_name': 'GS01234 Steam temperature',
      'icon': 'mdi:kettle-steam',
      'max': 131,
      'min': 126,
      'mode': <NumberMode.AUTO: 'auto'>,
      'step': 0.1,
      'unit_of_measurement': <UnitOfTemperature.CELSIUS: '°C'>,
    }),
    'context': <ANY>,
    'entity_id': 'number.gs01234_steam_temperature',
    'last_changed': <ANY>,
    'last_updated': <ANY>,
    'state': '128',
  })
# ---
# name: test_steam_boiler[GS3 MP].1
  EntityRegistryEntrySnapshot({
    'aliases': set({
    }),
    'area_id': None,
    'capabilities': dict({
      'max': 131,
      'min': 126,
      'mode': <NumberMode.AUTO: 'auto'>,
      'step': 0.1,
    }),
    'config_entry_id': <ANY>,
    'device_class': None,
    'device_id': <ANY>,
    'disabled_by': None,
    'domain': 'number',
    'entity_category': None,
    'entity_id': 'number.gs01234_steam_temperature',
    'has_entity_name': True,
    'hidden_by': None,
    'icon': None,
    'id': <ANY>,
    'name': None,
    'options': dict({
    }),
    'original_device_class': <NumberDeviceClass.TEMPERATURE: 'temperature'>,
    'original_icon': 'mdi:kettle-steam',
    'original_name': 'Steam temperature',
    'platform': 'lamarzocco',
    'previous_unique_id': None,
    'supported_features': 0,
    'translation_key': 'steam_temp',
    'unique_id': 'GS01234_steam_temp',
    'unit_of_measurement': <UnitOfTemperature.CELSIUS: '°C'>,
  })
# ---
